enrollment-server/src/main/java/com/wultra/app/enrollmentserver/impl/service/OperationTemplateService.java

```diff
@@ -44,20 +46,36 @@ public OperationTemplateService(OperationTemplateRepository operationTemplateRep
     }
 
     /**
-     * Find the operation template for the given type and language. Falling back to EN locale.
+     * Find the operation template for the given type and language.
+     * <p>
+     * Falling back to EN locale and later on to any found language.
      *
      * @param operationType Operation type.
      * @param language Template language.
      * @return Found operation template or empty.
      */
     public Optional<OperationTemplateEntity> findTemplate(@NotNull String operationType, @NotNull String language) {
-        return operationTemplateRepository.findFirstByLanguageAndPlaceholder(language, operationType).or(() -> {
+        return operationTemplateRepository.findFirstByLanguageAndPlaceholder(language, operationType).or(() ->
+                findTemplateFallback(operationType, language));
+    }
+
+    private Optional<OperationTemplateEntity> findTemplateFallback(final String operationType, final String language) {
+        if (!DEFAULT_LANGUAGE.equals(language)) {
             logger.debug("Trying fallback to EN locale for operationType={}", operationType);
-            return operationTemplateRepository.findFirstByLanguageAndPlaceholder("en", operationType).or(() -> {
-                logger.debug("Trying fallback to any locale for operationType={}", operationType);
-                return operationTemplateRepository.findFirstByPlaceholder(operationType);
-            });
-        });
+            return findEnglishTemplate(operationType);
+        } else {
+            return findAnyTemplate(operationType);
+        }
+    }
+
+    private Optional<OperationTemplateEntity> findEnglishTemplate(final String operationType) {
```

Comment:
Can we call this method `findDefaultTemplate` instead of `findEnglishTemplate`? The variable name is `DEFAULT_LANGUAGE`, so it will correspond nicely.